remoto/tests/test_process.py

```diff
@@ -38,3 +38,15 @@ def test_extend_env_gets_removed(self):
         fake_conn.receive.return_value = {'PATH': '/home/alfredo/bin'}
         result = process.extend_env(fake_conn, {'extend_env': {'CEPH_VOLUME_DEBUG': '1'}})
         assert result.get('extend_env') is None
+
+    def test_remote_check_command_encoding_returns_bytes(self):
+        fake_conn = Mock()
+        fake_conn.gateway.remote_exec.return_value = fake_conn
+        fake_conn.receive.return_value = {}
+        test_str = "remote check test"
+        test_str_bytes = "remote test check".encode('utf-8')
+
+        assert process._remote_check(fake_conn,cmd="/bin/echo", stdin=test_str) is None
+        assert process._remote_check(fake_conn,cmd="/bin/echo", stdin=test_str_bytes) is None
```

Comment:
This test is almost there. What this should be checking is that a plain string is used as the `stdin` argument and then it gets converted. Did you try to patch `subprocess` here so that we can tell if the `stdin` that gets passed into `communicate()` is encoded?

Reply:
Sure, rewrote to patch out the `subprocess` and check what gets passed into `communicate`